cassiopeia/plotting/itol_utilities.py

```diff
@@ -0,0 +1,623 @@
+
+    file_format = export_filepath.split("/")[-1].split(".")[-1]
+
+    if file_format not in ["png", "svg", "eps", "ps", "pdf"]:
+        raise iTOLError(
+            "File format must be one of " "'png', 'svg', 'eps', 'ps', 'pdf']"
+        )
+
+    itol_uploader = Itol()
+    itol_uploader.add_file(os.path.join(temporary_directory, "tree_to_plot.tree"))
+
+    files = []
+    if allele_table is not None:
+        files += create_indel_heatmap(
+            allele_table,
+            cassiopeia_tree,
+            f"{tree_name}.allele",
+            temporary_directory,
+            indel_colors,
+            indel_priors,
+            random_state,
+        )
+
+    for meta_item in meta_data:
+        if meta_item not in cassiopeia_tree.cell_meta.columns:
+            raise iTOLError("Meta data item not in CassiopeiaTree cell meta.")
+
+        values = cassiopeia_tree.cell_meta[meta_item]
+
+        if pd.api.types.is_numeric_dtype(values):
+            files.append(create_gradient_from_df(
+                values, cassiopeia_tree, f"{tree_name}.{meta_item}", temporary_directory,
+            ))
+
+        if pd.api.types.is_string_dtype(values):
+            colors = palette[: len(values.unique())]
+            colors = [hex_to_rgb(color) for color in colors]
+            colormap = dict(zip(np.unique(values), colors))
+
+            files.append(create_colorbar(
+                values,
+                cassiopeia_tree,
+                colormap,
+                f"{tree_name}.{meta_item}",
+                temporary_directory,
+                create_legend=include_legend,
+            ))
+
+    for _file in files:
+        itol_uploader.add_file(_file)
+
+    itol_uploader.params["treeName"] = tree_name
+    itol_uploader.params["APIkey"] = api_key
+    itol_uploader.params["projectName"] = project_name
+
+    good_upload = itol_uploader.upload()
+    if not good_upload:
+        raise iTOLError(itol_uploader.comm.upload_output)
+
+    print("iTOL output: " + str(itol_uploader.comm.upload_output))
+    print("Tree Web Page URL: " + itol_uploader.get_webpage())
+    print("Warnings: " + str(itol_uploader.comm.warnings))
+
+    tree_id = itol_uploader.comm.tree_id
+
+    itol_exporter = ItolExport()
+
+    # set parameters:
+    itol_exporter.set_export_param_value("tree", tree_id)
+    itol_exporter.set_export_param_value("format", file_format)
+    if rect:
+        # rectangular tree settings
+        itol_exporter.set_export_param_value("display_mode", 1)
+    else:
+        # circular tree settings
+        itol_exporter.set_export_param_value("display_mode", 2)
+        itol_exporter.set_export_param_value("arc", 359)
+        itol_exporter.set_export_param_value("rotation", 270)
+
+    itol_exporter.set_export_param_value("leaf_sorting", 1)
+    itol_exporter.set_export_param_value("label_display", 0)
+    itol_exporter.set_export_param_value("internal_marks", 0)
+    itol_exporter.set_export_param_value("ignore_branch_length", 1)
```

Comment:
How easy is it to plot the branch lengths or node ages too? Can we make this a flag?

Reply:
Yeah, this is a good idea -- I'll add an argument for taking into account branch lengths.